### DISCUSSION TITLE
Suggested configuration for non-programmers?

### DISCUSSION THREAD

**Discussion by BrianZbr** [company-mode/company-mode#1170] [Q&A] [answered]
My coding skills are minimal, but I use emacs for writing papers in org-mode. I'm looking for an auto-completion feature that will help me type faster and more accurately, similar to what is typically found on a smartphone. I posted [this question](http://emacs.stackexchange.com/questions/18304/optimal-settings-for-auto-complete-for-writing-papers-prose-in-natural-languag/18330) on the Emacs Stackexchange, which is what led me to `company-mode`.

Now I've played with company-mode a bit and I've read the basic documentation. My sense is that company-mode could be very useful to me if I could get the backend and the other settings right. But that would be very challenging for me to try and figure out on my own. 

So if you can suggest a configuration which is optimized for writing, I would greatly appreciate it. 

Here are some relevant thoughts based on my experience with company-mode so far. I'm not sure if these things can be easily addressed through the right configuration, or not. Let me know if any of these ideas may warrant a feature request and I will make one.
- Hippie-expand seems to do a slightly better job finding candidates for me. Here is the function list I have it set with: `(try-expand-all-abbrevs try-expand-dabbrev try-expand-dabbrev-all-buffers try-expand-dabbrev-from-kill try-complete-file-name-partially try-complete-file-name try-expand-list try-expand-line try-complete-lisp-symbol-partially try-complete-lisp-symbol)` 
- I'd like to have the option of calling up a list of candidates before the idle timer runs out with a keybinding like <kbd>M-/</kbd>
- If I'm not mistaken, whenever company-mode finds exactly one _single_ candidate, then it shows that candidate with underlining instead of opening a menu. If I could pull up the full list of candidates as just described, then I would always want the _top_ candidate to appear that same way as soon as it is found, even if there are other candidates.
- If there is a candidate which can be found _both_ through `dabbrev` _and_ with `ispell`, that should probably take precedence over a candidate which can only be found in `dabbrev`, because it's likely to be misspelled. 
- I do not normally want candidates with numerical digits and/or special characters like `%` or `$`. It is probably OK to hide those candidates from me completely. 
- If I have already typed the complete word and it is correctly spelled, then I only want to see candidates that are _longer_, and are _valid_ words (like, according to ispell). So, if I have typed "contact", then "contacted", "contacting", etc. are candidates but not "contact" itself or anything like "contact.", "contact'", etc. 
- A little bit of "fuzziness" would help a lot, but not too much. The key thing is, if I'm misspelling a word, company-mode should still help me.

**Comment by dgutov** [comment, marked as answer]
Hi there!

Overall, you can get some of the items in the list, but others seem to be calling for a new backend, which would perform spelling corrections and do dictionary-based sorting and filtering. Maybe you should ask on Reddit or on IRC if someone would like to write a backend like that. My own interests lie primarily in code completion.

> Hippie-expand seems to do a slightly better job finding candidates for me.

You can still have a separate binding for it.

> I'd like to have the option of calling up a list of candidates before the idle timer runs out

``` el
(global-set-key (kbd "C-/") 'company-complete)
```

> If I could pull up the full list of candidates as just described, then I would always want the top candidate to appear that same way as soon as it is found, even if there are other candidates.

Replace `company-preview-if-just-one-frontend` with `company-preview` in `company-frontends`.

> I do not normally want candidates with numerical digits and/or special characters like % or $.

Add a function that would remove them, to `company-transformers`.

**Comment by BrianZbr** [comment]
OK, thank you for the information... I'll close the issue for now.

**Comment by yugaego** [comment]
A recently added [User Manual](https://company-mode.github.io/manual/) may also help non-programmers by introducing the main concepts, structure, and usage patterns of the package.
